web/modules/custom/joinup_migrate/src/Plugin/migrate/source/FileInline.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Drupal\joinup_migrate\Plugin\migrate\source;
+
+use Drupal\Core\Database\Database;
+use Drupal\migrate\Plugin\migrate\source\SourcePluginBase;
+use Drupal\migrate\Row;
+
+/**
+ * Source plugin for inline files.
+ *
+ * @MigrateSource(
+ *   id = "file_inline"
+ * )
+ */
+class FileInline extends SourcePluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getIds() {
+    return ['fid' => ['type' => 'string']];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function fields() {
+    return [
+      'fid' => $this->t('File ID'),
+      'path' => $this->t('File path'),
+      'timestamp' => $this->t('Created time'),
+      'uid' => $this->t('File owner'),
+      'destination_uri' => $this->t('Destination URI'),
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function initializeIterator() {
+    $rows = [];
+    $timestamp = \Drupal::time()->getRequestTime();
+
+    foreach ($this->getInlineFiles() as $file) {
+      list($type, $basename) = explode('/', $file, 2);
+      $path = "ckeditor_files/$file";
+      $rows[] = [
+        'fid' => $path,
+        'path' => $path,
+        'timestamp' => $timestamp,
+        'uid' => 1,
+        'destination_uri' => "public://inline-$type/$basename",
+      ];
+    }
+
+    return new \ArrayIterator($rows);
+  }
+
+  /**
+   * Provides a list of inline files.
+   *
+   * @return string[]
+   *   A list of inline files paths. The path is relative to Drupal 6 CKEditor
+   *   files directory 'sites/default/files/ckeditor_files'.
+   */
+  protected function getInlineFiles() {
+    $db = Database::getConnection('default', 'migrate');
+    $search = ['img' => 'src', 'a' => 'href'];
+
+    $files = [];
+    foreach (static::$bodyFields as $table => $fields) {
+      $items = $db->select($table)->fields($table, $fields)->execute()->fetchAll();
```

Comment:
This could go wrong if the set is big, we might do a while ($item = $query->fetchRow())